src/SFML/Audio/MiniaudioUtils.cpp

```diff
@@ -0,0 +1,264 @@
+}
+
+
+////////////////////////////////////////////////////////////
+ma_channel MiniaudioUtils::soundChannelToMiniaudioChannel(sf::SoundChannel soundChannel)
+{
+    switch (soundChannel)
+    {
+        case SoundChannel::Unspecified:
+            return MA_CHANNEL_NONE;
+        case SoundChannel::Mono:
+            return MA_CHANNEL_MONO;
+        case SoundChannel::FrontLeft:
+            return MA_CHANNEL_FRONT_LEFT;
+        case SoundChannel::FrontRight:
+            return MA_CHANNEL_FRONT_RIGHT;
+        case SoundChannel::FrontCenter:
+            return MA_CHANNEL_FRONT_CENTER;
+        case SoundChannel::FrontLeftOfCenter:
+            return MA_CHANNEL_FRONT_LEFT_CENTER;
+        case SoundChannel::FrontRightOfCenter:
+            return MA_CHANNEL_FRONT_RIGHT_CENTER;
+        case SoundChannel::LowFrequencyEffects:
+            return MA_CHANNEL_LFE;
+        case SoundChannel::BackLeft:
+            return MA_CHANNEL_BACK_LEFT;
+        case SoundChannel::BackRight:
+            return MA_CHANNEL_BACK_RIGHT;
+        case SoundChannel::BackCenter:
+            return MA_CHANNEL_BACK_CENTER;
+        case SoundChannel::SideLeft:
+            return MA_CHANNEL_SIDE_LEFT;
+        case SoundChannel::SideRight:
+            return MA_CHANNEL_SIDE_RIGHT;
+        case SoundChannel::TopCenter:
+            return MA_CHANNEL_TOP_CENTER;
+        case SoundChannel::TopFrontLeft:
+            return MA_CHANNEL_TOP_FRONT_LEFT;
+        case SoundChannel::TopFrontRight:
+            return MA_CHANNEL_TOP_FRONT_RIGHT;
+        case SoundChannel::TopFrontCenter:
+            return MA_CHANNEL_TOP_FRONT_CENTER;
+        case SoundChannel::TopBackLeft:
+            return MA_CHANNEL_TOP_BACK_LEFT;
+        case SoundChannel::TopBackRight:
+            return MA_CHANNEL_TOP_BACK_RIGHT;
+        default:
+            assert(soundChannel == SoundChannel::TopBackCenter);
+            return MA_CHANNEL_TOP_BACK_CENTER;
+    }
+}
+
+
+////////////////////////////////////////////////////////////
+void MiniaudioUtils::initializeSoundWithDefaultSettings(ma_sound& sound)
+{
+    ma_sound_set_pitch(&sound, 1.f);
+    ma_sound_set_pan(&sound, 0.f);
+    ma_sound_set_volume(&sound, 1.f);
+    ma_sound_set_spatialization_enabled(&sound, MA_TRUE);
+    ma_sound_set_position(&sound, 0.f, 0.f, 0.f);
+    ma_sound_set_direction(&sound, 0.f, 0.f, -1.f);
+    ma_sound_set_cone(&sound, degrees(360).asRadians(), degrees(360).asRadians(), 0.f); // inner = 360 degrees, outer = 360 degrees, gain = 0
+    ma_sound_set_directional_attenuation_factor(&sound, 1.f);
+    ma_sound_set_velocity(&sound, 0.f, 0.f, 0.f);
+    ma_sound_set_doppler_factor(&sound, 1.f);
+    ma_sound_set_positioning(&sound, ma_positioning_absolute);
+    ma_sound_set_min_distance(&sound, 1.f);
+    ma_sound_set_max_distance(&sound, std::numeric_limits<float>::max());
+    ma_sound_set_min_gain(&sound, 0.f);
+    ma_sound_set_max_gain(&sound, 1.f);
+    ma_sound_set_rolloff(&sound, 1.f);
```

Comment:
A simpler option would be to provide default values to the data in `SavedSettings`. Then this function can be implanted as one line that simply calls `applySettings` with a default-constructed `SavedSettings`.

Reply:
https://github.com/SFML/SFML/compare/feature/miniaudio...saved_settings

Here's what that looks like. What do you think?

Reply:
Implemented.